lib/run-tests/babel-agent.js

```diff
@@ -39,7 +48,68 @@ class BabelAgent extends NodeAgent {
   }
 }
 
-BabelAgent.runtime = fs.readFileSync(__dirname + "/babel-agent-runtime.js", "utf8")
+BabelAgent.runtime = fs
+  .readFileSync(__dirname + "/babel-agent-runtime.js", "utf8")
   .replace(/\$POLYFILLS/g, POLYFILLS);
 
-module.exports = BabelAgent;
+//module.exports = BabelAgent;
+
+const TEST_TIMEOUT = 60 * 1000; // 1 minute
+
+function timeout(file, waitMs = TEST_TIMEOUT) {
+  return new Promise((_resolve, reject) =>
+    setTimeout(
+      () => reject(new Error(`test ${file} timed out after ${waitMs} ms`)),
+      waitMs
+    )
+  );
+}
+
+let agent;
+
+module.exports = {
+  runTest: async function (opts, test) {
+    if (!agent) agent = new BabelAgent(opts);
+
+    let { attrs, contents, file } = test;
+    const isModule = attrs.flags.module;
+
+    try {
+      contents = transpile(contents, attrs.features, isModule);
+    } catch (error) {
+      return { result: "parser error", error };
+    }
+
+    const hash = crypto.createHash("sha1").update(contents).digest("hex");
+
+    const cached = cache.get(hash);
+    if (cached) return cached;
+
+    let result;
+    let ret;
+    try {
+      result = await Promise.race([
+        agent.evalScript({
+          attrs,
+          contents,
+          file: path.join(opts.testRoot, file),
+        }),
+        timeout(file),
+      ]);
+    } catch (error) {
+      agent.stop(); // kill process avoid 100% cpu usage
```

Comment:
Not sure if it has something to do with the slowness in CI, but on my computer several infinite looping processes are taking very, very, very much cpu.😅